crates/chain/src/keychain/txout_index.rs

```diff
@@ -112,22 +122,24 @@ const DEFAULT_LOOKAHEAD: u32 = 25;
 /// above the last revealed index. These additionally-derived script pubkeys are called the
 /// lookahead.
 ///
-/// The [`KeychainTxOutIndex`] is constructed with the `lookahead` and cannot be altered. The
-/// default `lookahead` count is 1000. Use [`new`] to set a custom `lookahead`.
+/// The [`KeychainTxOutIndex`] is constructed with the `lookahead` and cannot be altered. See
+/// [`DEFAULT_LOOKAHEAD`] for the value used in the `Default` implementation. Use [`new`] to set a
+/// custom `lookahead`.
 ///
 /// # Unbounded script pubkey iterator
 ///
 /// For script-pubkey-based chain sources (such as Electrum/Esplora), an initial scan is best done
 /// by iterating though derived script pubkeys one by one and requesting transaction histories for
 /// each script pubkey. We will stop after x-number of script pubkeys have empty histories. An
-/// unbounded script pubkey iterator is useful to pass to such a chain source.
+/// unbounded script pubkey iterator is useful to pass to such a chain source because it doesn't
+/// require holding a reference to the index.
 ///
 /// Call [`unbounded_spk_iter`] to get an unbounded script pubkey iterator for a given keychain.
 /// Call [`all_unbounded_spk_iters`] to get unbounded script pubkey iterators for all keychains.
 ///
 /// # Change sets
 ///
-/// Methods that can update the last revealed index or add keychains will return [`super::ChangeSet`] to report
+/// Methods that can update the last revealed index or add keychains will return [`ChangeSet`] to report
 /// these changes. This can be persisted for future recovery.
```

Comment:
⛏️  Would be good to make this recommendation stronger here.
```suggestion
/// these changes. This [`ChangeSet`] should be persisted for future recovery.
```